apps/election-manager/src/lib/votecounting.ts

```diff
@@ -671,29 +679,29 @@ export function filterTalliesByParams(
       getEmptyTally()
     : undefined
 
-  // TODO(#2975): Once we're removing duplicate ballots, make finding the intersection of these lists more performant.
   const ballotCountsByVotingMethod: Dictionary<number> = {}
   Object.values(VotingMethod).forEach(
     (votingMethod) => (ballotCountsByVotingMethod[votingMethod] = 0)
   )
-  overallTally.castVoteRecords.forEach((CVR) => {
-    if (!precinctTally || precinctTally.castVoteRecords.includes(CVR)) {
-      if (!scannerTally || scannerTally.castVoteRecords.includes(CVR)) {
-        if (!partyTally || partyTally.castVoteRecords.includes(CVR)) {
+  for (const _cvrId in overallTally.castVoteRecords) {
+    const CVR = overallTally.castVoteRecords[_cvrId]!
+    if (!precinctTally || precinctTally.castVoteRecords[_cvrId] !== undefined) {
+      if (!scannerTally || scannerTally.castVoteRecords[_cvrId] !== undefined) {
+        if (!partyTally || partyTally.castVoteRecords[_cvrId] !== undefined) {
```

Comment:
Is this the crux of the performance gains you observed? Could this have been achieved with a `Set<CastVoteRecord>`?